stonesoup/models/base.py

```diff
@@ -43,16 +93,17 @@ class LinearModel(Model):
     Base/Abstract class for all linear models"""
 
     @abstractmethod
-    def matrix(self):
-        """ Model matrix"""
-        pass
+    def matrix(self, **kwargs) -> np.ndarray:
+        """Model matrix"""
+        raise NotImplementedError
 
-    def function(self, state, noise=False, **kwargs):
+    def function(self, state: State, noise: Union[bool, np.ndarray] = False,
+                 **kwargs) -> Union[StateVector, StateVectors]:
         """Model linear function :math:`f_k(x(k),w(k)) = F_k(x_k) + w_k`
 
         Parameters
         ----------
-        state: :class:`~.State`
+        state: State
```

Comment:
why is this changed?

Reply:
NumpyDoc assumes it's a class when no spaces, so extra `:class:` wrapper is redundant.